### PR TITLE
fix: distribute test files to shards more evenly

### PR DESCRIPTION
**Pull request by Shinyaigeek**
### Description

Fixes: #8244

In current implementation, test files are distributed with `Math.ceil(files.length / count)` in `{index}/{count}` shard setting.
This causes #8244 because this ceil based distribution logic can consume all test files by shards without the ending shard.
In addition to this, I believe this logic causes a bit lower performance in an edge-case like:

- test files: 10
- `--shard index/3`

In the above case, test files will be distributed as `[3, 3, 3, 1]` but I think it is more ideal to be distributed as `[3, 3, 2, 2]` in the point of parallelism.

This PR fixes this issue by introducing distribution logic based on `Math.floor(files.length / count)` instead of `Math.ceil`

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: test/config/test/shard.test.ts [hunk 1]
```diff
@@ -40,6 +40,52 @@ test('--shard=2/2', async () => {
...
+test('--shard=3/3 should distribute files evenly', async () => {
+  const { stdout } = await runVitest({ shard: '3/3' })
+
+  const paths = parsePaths(stdout)
+
+  // With 3 files and 3 shards, should get 1 file per shard
+  expect(paths).toEqual(['3.test.js'])
+})
+
+test('4 files with 3 shards should distribute as [2,1,1] not [2,2,0]', async () => {
```
Comment (AriPerkkio) [inline test/config/test/shard.test.ts]:
```suggestion
test('4 files with 3 shards should distribute evenly', async () => {
```

**Review by AriPerkkio — APPROVED** [review]
Looks good, thanks @Shinyaigeek! 💯
